File: StarLegacy/src/main/kotlin/net/starlegacy/feature/customitem/BatteryItems.java
package net.starlegacy.feature.customitem

import org.bukkit.Material;

class BatteryItems {
	companion object {
		private fun registerBattery(name: String, model: Int, power: Int): BatteryItem{
			val item = BatteryItem(
					id = "battery_${name.stripColor().lowercase().replace(" ", "_")}",
					displayName = name,
					material = Material.APPLE,
					model = model,
					maxPower = power,
			)
			CustomItemManager.register(item)
			return item
		}

		fun register() {
			registerBattery("Battery ${RED}A", 7, 500)
			registerBattery("Battery ${GREEN}M", 8, 1000)
			registerBattery("Battery ${GOLD}G", 9, 2000)
		}
	}
}

File: StarLegacy/src/main/kotlin/net/starlegacy/feature/customitem/BatteryItems.kt
package net.starlegacy.feature.customitem

import net.starlegacy.feature.customitem.type.BatteryItem
import net.starlegacy.util.SLTextStyle
import net.starlegacy.util.stripColor
import org.bukkit.Material

object BatteryItems {

	private fun registerBattery(name: String, model: Int, power: Int): BatteryItem {
		val item = BatteryItem(
			id = "battery_${name.stripColor().lowercase().replace(" ", "_")}",
			displayName = name,
			material = Material.APPLE,
			model = model,
			maxPower = power,
		)
		CustomItemManager.register(item)
		return item
	}

	fun register() {
		registerBattery("Battery ${SLTextStyle.RED}A", 7, 500)
		registerBattery("Battery ${SLTextStyle.GREEN}M", 8, 1000)
		registerBattery("Battery ${SLTextStyle.GOLD}G", 9, 2000)
	}

}

